connexion/security.py

```diff
@@ -403,6 +523,126 @@ async def wrapper(request, token, required_scopes):
 
         return wrapper
 
+
+class OIDCSecurityHandlerFactory(AbstractSecurityHandlerFactory):
+    """
+    SecurityHandlerFactory for the OIDC security scheme.
+    """
+
+    # TODO
+
+
+SECURITY_HANDLERS = {
+    # Swagger 2: `type: basic`
+    # OpenAPI 3: `type: http` and `scheme: basic`
+    "basic": BasicSecurityHandlerFactory,
+    # Swagger 2 and OpenAPI 3
+    "apiKey": ApiKeySecurityHandlerFactory,
+    "oauth2": OAuthSecurityHandlerFactory,
+    # OpenAPI 3 only
+    "openIdConnect": OIDCSecurityHandlerFactory,
+    # OpenAPI 3: http schemes
+    "bearer": BearerSecurityHandlerFactory,
+}
+
+
+# TODO: Split up
+#   along security schemes
+#   along 'functionality' -> get_*_func (resolver), security_{deny,passthrough}, and verify_*
+class SecurityHandlerFactory(StandardHandlerFactory):
```

Comment:
Seems like we're now calling two distinct things `SecurityHandlerFactory`, this class, and the classes it calls. We should probably rename one of them for clarity.

Reply:
Good point, let's keep this one as `SecurityHandlerFactory` and rename the other ones.

Reply:
Noticed that it is still subclassing from the `StandardHandlerFactory` - have to rework this as well. Not sure if these standard functions need to be able to be overwritten/customized.